shell/platform/linux/fl_platform_views_plugin.cc

```diff
@@ -0,0 +1,310 @@
+static constexpr char kDisposeMethod[] = "dispose";
+static constexpr char kAcceptGestureMethod[] = "acceptGesture";
+static constexpr char kRejectGestureMethod[] = "rejectGesture";
+static constexpr char kEnterMethod[] = "enter";
+static constexpr char kExitMethod[] = "exit";
+
+struct _FlPlatformViewsPlugin {
+  GObject parent_instance;
+
+  FlMethodChannel* channel;
+
+  FlGestureHelper* gesture_helper;
+
+  GHashTable* factories;  // string -> FlPlatformViewFactory*
+
+  GHashTable* platform_views;  // int -> FlPlatformView*
+};
+
+G_DEFINE_TYPE(FlPlatformViewsPlugin, fl_platform_views_plugin, G_TYPE_OBJECT)
+
+// Sends the method call response to Flutter.
+static void send_response(FlMethodCall* method_call,
+                          FlMethodResponse* response) {
+  g_autoptr(GError) error = nullptr;
+  if (!fl_method_call_respond(method_call, response, &error)) {
+    g_warning("Failed to send method call response: %s", error->message);
+  }
+}
+
+static FlPlatformView* get_platform_view(FlPlatformViewsPlugin* self,
+                                         int64_t id) {
+  gpointer p = g_hash_table_lookup(self->platform_views, GINT_TO_POINTER(id));
+  if (p && FL_IS_PLATFORM_VIEW(p)) {
+    return FL_PLATFORM_VIEW(p);
+  } else {
+    return nullptr;
+  }
+}
+
+// Creates a new platform view.
+static FlMethodResponse* platform_views_create(FlPlatformViewsPlugin* self,
+                                               FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_MAP) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument map missing or malformed", nullptr));
+  }
+
+  FlValue* id_value = fl_value_lookup_string(args, "id");
+  if (id_value == nullptr || fl_value_get_type(id_value) != FL_VALUE_TYPE_INT) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view id", nullptr));
+  }
+  int64_t id = fl_value_get_int(id_value);
+
+  FlValue* view_type_value = fl_value_lookup_string(args, "viewType");
+  if (view_type_value == nullptr ||
+      fl_value_get_type(view_type_value) != FL_VALUE_TYPE_STRING) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view viewType", nullptr));
+  }
+  const gchar* view_type = fl_value_get_string(view_type_value);
+
+  FlValue* params_value = fl_value_lookup_string(args, "params");
+
+  FlPlatformView* platform_view = get_platform_view(self, id);
+  if (platform_view) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Platform view id already exists", nullptr));
+  }
+
+  gpointer factory_pointer = g_hash_table_lookup(self->factories, view_type);
+  if (!factory_pointer) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "View type does not exist", nullptr));
+  }
+  FlPlatformViewFactory* factory = FL_PLATFORM_VIEW_FACTORY(factory_pointer);
```

Comment:
This probably won't fail, but it can't hurt to move the `FL_PLATFORM_VIEW_FACTORY()` cast to above the check and then check if `factory` exists, rather than checking `factory_pointer` (assuming that factory *shouldn't* ever be null)

Reply:
Done.